### PR TITLE
[AGM-863] How to debug filters

### LINKED ISSUE #328: Add "how to debug filters" section to the documentation
This section should explain what filtering log is (in every product) and explain some necessary details on the rules debugging.

For instance, `debug` mode of the scriptlet filters.

### PR DESCRIPTION
**Pull request by vbagirov**
Closes #328

### CODE REVIEW

File: 01.general/02.how-to-create-your-own-ad-filters/docs.en.md [hunk 1]
```diff
@@ -2150,6 +2122,66 @@ This rule will be available for every platform except Safari extension, iOS, and
...
+<a id="debug-filtering-log"></a>
+### Filtering log
+
+Filtering log is an advanced tool that will be helpful mostly to filter developers. It lists all web requests that pass through AdGuard, gives you exhaustive information on each of them, offers multiple sorting options, and has other useful features.
+
+Depending on which AdGuard product you're using, Filtering log can be located in different places. 
+
+* In **AdGuard for Windows** you'll find it inside *Ad Blocker* tab or via the tray menu;
+* In **AdGuard for Mac** it's under *Settings > Advanced > Filtering log*;
+* In **AdGuard for Android** it's a separate item in the side menu;
```
Comment (Alex-302) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:2140]: and in Assistant, the log for specific app or site.

File: 01.general/02.how-to-create-your-own-ad-filters/docs.en.md [hunk 2]
```diff
@@ -2150,6 +2122,66 @@ This rule will be available for every platform except Safari extension, iOS, and
...
+* In **AdGuard browser extensions** it's accessible from the *Miscellaneous* settings tab or by right-clicking the extension icon.
```
Comment (Alex-302) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:2141]: Only Chromium and Firefox based browsers filtering log show applied 
- element hiding rules(including CSS, ExtCSS)
- JS and scriptlets.

File: 01.general/02.how-to-create-your-own-ad-filters/docs.en.md [hunk 3]
```diff
@@ -2150,6 +2122,66 @@ This rule will be available for every platform except Safari extension, iOS, and
...
+
+> In **AdGuard for iOS** and in **AdGuard for Safari** Filtering log does not exist because of the way content blockers are implemented in Safari. AdGuard doesn't see the web requests and therefore can't display them.
+
+<a id="selectors-debugging-mode"></a>
+### Selectors debugging mode
+
+Sometimes, you might need to check the performance of a given selector or a stylesheet. In order to do it without interacting with javascript directly, you can use a special `debug` style property. When `ExtendedCss` meets this property, it enables the "debug"-mode either for a single selector or for all selectors depending on the `debug` value.
+
+**Debugging a single selector**
```
Comment (Alex-302) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:2150]: describe debugging

File: 01.general/02.how-to-create-your-own-ad-filters/docs.en.md [hunk 4]
```diff
@@ -2150,6 +2122,66 @@ This rule will be available for every platform except Safari extension, iOS, and
...
+```
+#$#.banner { display: none; debug: true; }
```
Comment (Alex-302) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:2152]: `#$?#`

File: 01.general/02.how-to-create-your-own-ad-filters/docs.en.md [hunk 5]
```diff
@@ -2150,6 +2122,66 @@ This rule will be available for every platform except Safari extension, iOS, and
...
+```
+
+**Enabling global debug**
+```
+#$#.banner { display: none; debug: global; }
```
Comment (Alex-302) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:2157]: `#$?#`

**Review by Alex-302 — CHANGES_REQUESTED** [review]
Also add examples of debugging Extended CSS

File: 01.general/02.how-to-create-your-own-ad-filters/docs.en.md [hunk 6]
```diff
@@ -2150,6 +2122,66 @@ This rule will be available for every platform except Safari extension, iOS, and
...
+```
+var selectorText = "div.block[-ext-has='.header:matches-css-after(content: Anzeige)']";
+
+ExtendedCss.query(selectorText) // returns an array of Elements matching selectorText
+```
+
+<a id="debug-scriptlets"></a>
+### Debugging scriptlets
+
+If you're using AdGuard browser extension and want to debug a [scriptlet rule](https://kb.adguard.com/en/general/how-to-create-your-own-ad-filters#scriptlets), you can get additional information by simpy having the Filtering log opened. In that case, scriptlets will switch to debug mode and will write more information to the browser's console.
```
Comment (Sergey-Lyapin) [inline 01.general/02.how-to-create-your-own-ad-filters/docs.en.md:2182]: Maybe we should add a more detailed description of debugging with scriptlets and add examples?

For example, these scriptlets can be useful for debugging:

```
debug-current-inline-script
debug-on-property-read
debug-on-property-write
log-addEventListener
log-eval
log
```

**Review by slavaleleka — CHANGES_REQUESTED** [review]
[requestAnimationFrame](https://github.com/AdguardTeam/Scriptlets/blob/master/wiki/about-scriptlets.md#prevent-requestanimationframe), [prevent-setInterval](https://github.com/AdguardTeam/Scriptlets/blob/master/wiki/about-scriptlets.md#prevent-setinterval) and [prevent-setTimeout](https://github.com/AdguardTeam/Scriptlets/blob/master/wiki/about-scriptlets.md#prevent-settimeout) can be used without parameters for debug purposes

**Review by slavaleleka — DISMISSED** [review]

**Review by slavaleleka — APPROVED** [review]

**Pull request merged**